index.js

```diff
@@ -320,12 +337,10 @@
   //  xor :: a -> a -> a
   S.xor = R.curry(function(x, y) {
     assertTypeMatch(x, y);
-    switch (true) {
-      case toBoolean(x) !== toBoolean(y): return or(x, y);
-      case R.is(Array, x):                return [];
-      case R.is(Boolean, x):              return false;
-      default:                            return x.empty();
-    }
+    var xBool = toBoolean(x);
+    var yBool = toBoolean(y);
+    var xEmpty = empty(x);
+    return xBool !== yBool ? or(x, y) : xEmpty;
```

Comment:
It may be worth leaving a comment here explaining that empty() has the intended side effecting of throwing a `TypeError`.

Also, what's special about the empty value of the first element? Should you also verify that the second element has an empty value?